### PR TITLE
Support Bazel build

### PR DESCRIPTION
**Pull request by adamyi**
Support building with [Bazel](https://bazel.build).

To build it, run
```
bazel build //bin/...
```

The built artifact will be at `bazel-bin/bin/protoc-gen-ts`.

This would facilitate supporting Protobuf in Bazel-built ts projects. This CL is made on top of the original WIP but seemingly abandoned PR #24.

/cc @pcj @MarcusLongmuir

### CODE REVIEW

**Comment by jonnyreeves** [comment]
Thanks for your submission, we also make use of Bazel at improbable 😀

Please can you modify the Travis build configuration to ensure that the hazel build is also executed?

Thanks

**Comment by adamyi** [comment]
@jonnyreeves done

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,12 @@
 .vscode
 node_modules
 node-debug.log
+package-lock.json
```
Comment (jonnyreeves) [inline .gitignore]: This change is unrelated, please revert

Comment (adamyi) [reply .gitignore]: Done

**Comment by adamyi** [comment]
Got Travis CI working now.

File: package.json [hunk 2]
```diff
@@ -33,12 +33,11 @@
     "@types/google-protobuf": "^3.2.4",
     "@types/mocha": "^2.2.40",
     "@types/node": "^7.0.5",
-    "@bazel/typescript": "^0.0.10",
     "babel": "^6.5.2",
     "chai": "^3.5.0",
     "mocha": "^3.2.0",
     "source-map-support": "^0.4.14",
-    "tslint": "^4.5.1",
+    "tslint": "^5.9.1",
```
Comment (adamyi) [inline package.json]: The old version would depend on an old version of tsutils, which is incompatible with the new Bazel typescript rules.

**Comment by jonny-improbable** [comment]
LGTM, sorry for the delay in merging. Thanks for your contributions @adamyi and @pcj
